File: unity_code/Actions/DelayAction.cs
using System;
namespace Assets.Scripts.Actions
{
    public class DelayAction : IGameAction
    {
        private readonly float targetTime;
        private float elapsedTime;

        public DelayAction (TimeSpan time)
        {
            this.targetTime = (float)time.TotalSeconds;
            this.elapsedTime = 0;
        }

        public ActionResult Step (float deltaTime)
        {
            if(elapsedTime > targetTime)
                return ActionResult.Finished;

            elapsedTime += deltaTime;
            return ActionResult.Continue;
        }
    }
}

File: src/main/ts/actions/DelayAction.ts
import { IGameAction } from './IGameAction';
import { ActionResult } from './ActionResult';

    export class DelayAction implements IGameAction
    {
        private targetTime:number;
        private elapsedTime:number;

        constructor(delay:number){
            this.targetTime = delay;
            this.elapsedTime = 0;
        }




        public step (deltaTime:number):ActionResult{
            this.elapsedTime += deltaTime;

            if(this.elapsedTime >= this.targetTime) {
                return ActionResult.Finished;
            }
            return ActionResult.Continue;
        }

    }


